### ISSUE TITLE
🐛 BUG: Proxying a websocket connection to a durable object (defined in a worker) leads to "invalid upgrade header" error in wrangler pages dev.

### ISSUE THREAD

**Issue by amadeuspagel** [cloudflare/workers-sdk#2676]
### Which Cloudflare product(s) does this pertain to?

Pages, Wrangler

### What version of `Wrangler` are you using?

2.9.0

### What operating system are you using?

Linux

### Describe the Bug

Proxying a websocket connection to a durable object (defined in a worker) leads to "invalid upgrade header" error in wrangler pages dev.

Replicate:

```
git clone https://github.com/amadeuspagel/pages-do-minimal
cd pages-do-minimal
npm run devworker
npm run devpages
```

then open localhost:8788

**Comment by third774** [comment]
Just ran the repro locally and this is what the error output looks like in the terminal
![image](https://user-images.githubusercontent.com/8732191/222272378-8100cab5-6b5c-4a5b-b878-83148dc5257d.png)

**Comment by CarmenPopoviciu** [comment]
Thank you both for your input. I can indeed confirm this is an issue and that the above repro is accurate. Seeing the same in my local repro as well.

We will look into it and hope to have a fix soon

**Comment by GregBrimble** [comment]
I _think_ I've got a fix in #2989. You can test it with `npm install --save-dev https://prerelease-registry.devprod.cloudflare.dev/workers-sdk/runs/4610496025/npm-package-wrangler-2989`.

I ran it against the reproduction provided by @amadeuspagel (thank you!) and now, rather than `invalid upgrade header`, we get `Web Socket request did not return status 101 Switching Protocols response with Web Socket` (which is expected given that [the Durable Object in that example doesn't actually return a WebSocket](https://github.com/amadeuspagel/pages-do-minimal/blob/main/worker/src/hello_durable.ts#L8)!)

I then tweaked that Durable Object to look like this:

```ts
import { Env } from ".";

export class HelloDurable implements DurableObject {
	constructor(
		private readonly state: DurableObjectState,
		private readonly env: Env
	) {}

	async fetch(request: Request): Promise<Response> {
		const [client, server] = Object.values(new WebSocketPair());
		server.accept();
		server.addEventListener("message", (event) => {
			client.send("echo: " + event.data);
		});
		return new Response(null, {
			status: 101,
			webSocket: client,
		});
	}
}
```

And sure enough, the WebSocket seemed to now establish itself correctly: `GET /pages_durable 101 Switching Protocols (13.05ms)`.

@third774, @CarmenPopoviciu , if y'all find a second to test against your own projects, that would be very helpful just to confirm :)

**Comment by CarmenPopoviciu** [comment]
@GregBrimble tested on my project and looks like the issue is fixed!
